### PR TITLE
Paystack payment integration

### PR DESCRIPTION
**Pull request by PinkCode15**

### CODE REVIEW

File: app/Actions/PlanPayment/ProcessWebhookAction.php [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+            logger($e);
+            return $this->formatApiResponse(500, 'Error occured', [], $e->getMessage());
+        }
+       
+    }
+
+
+    private function createUser($email, $pricingPlan, $amount) 
+    {
```
Comment (IDTitanium) [inline app/Actions/PlanPayment/ProcessWebhookAction.php:103]: This method should be in the action or service responsible for user registration.

File: app/Mail/SetPasswordMail.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    protected function generateSetPasswordUrl()
+    {
+        $token = Str::random(32);
+
+        PasswordReset::updateOrCreate(
+            ['email' => $this->data['user']->email],
+            [
+                'token' => $token,
+                'created_at' => now(),
```
Comment (IDTitanium) [inline app/Mail/SetPasswordMail.php:55]: There should expiry that is also set for this. Probably 24/48 hours.

**Review by IDTitanium — COMMENTED** [review]
General comment: Let's add types to all params and function return types. Nice work.

File: app/Actions/PlanPayment/ProcessWebhookAction.php (hunk 1 above)
Comment (PinkCode15) [reply app/Actions/PlanPayment/ProcessWebhookAction.php:103]: Yea, the major commonalities have been abstracted in the createNew() function. The extra things are unique to this

File: app/Mail/SetPasswordMail.php (hunk 2 above)
Comment (PinkCode15) [reply app/Mail/SetPasswordMail.php:55]: Yea, the table has a created_at field so when the request comes in it compares the current time to check if it has expired

**Comment by PinkCode15** [comment]
Thanks. Noted, I'll look into that

File: app/Mail/SetPasswordMail.php (hunk 2 above)
Comment (IDTitanium) [reply app/Mail/SetPasswordMail.php:55]: Alright that works.

File: app/Actions/PlanPayment/ProcessWebhookAction.php (hunk 1 above)
Comment (IDTitanium) [reply app/Actions/PlanPayment/ProcessWebhookAction.php:103]: Alrigth

**Review by IDTitanium — APPROVED** [review]

**Pull request merged**